core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -993,6 +999,12 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleApiVersionRequest(request: RequestChannel.Request) {
+    val responseHeader = new ResponseHeader(request.header.correlationId)
+    val responseBody = KafkaApis.apiVersionResponse
```

Comment:
This is related to the missed point in previous comment, should be fixed with it. So, yes it will need a check here for version, if higher than supported version is received, gladly send an error back :).